### PR TITLE
Fix assignment of return variables from modifiers.

### LINKED ISSUE #12061: [Sol->Yul] Function parameters not restored after modifier invocation
The following test case passes via IR and legacy (expectation is to fail via IR because output should be zero).

```
pragma solidity >= 0.0.0;
contract C0 {
  modifier m0(uint i1)
  {
        _;
  }
  function IRReturnsTwo() external m0(o0++) m0(o0++) returns (uint o0)
  {
  }
}
// ====
// compileViaYul: also
// ----
// IRReturnsTwo() -> 2
```

The following test passes on IR.

```
pragma solidity >= 0.0.0;
contract C0 {
  modifier m0(uint i1)
  {
        _;
  }
  function IRReturnsZero() external m0(o0++) m0(o0++) returns (uint o0, uint o1)
  {
  }
}
// ====
// compileViaYul: true
// ----
// IRReturnsZero() -> 0, 0
```

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/12061

### CODE REVIEW

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 1]
```diff
@@ -529,7 +529,7 @@ string IRGenerator::generateFunctionWithModifierInner(FunctionDefinition const&
 		for (size_t i = 0; i < retParams.size(); ++i)
 		{
 			retParamsIn.emplace_back(m_context.newYulVariable());
-			assignRetParams += retParams.back() + " := " + retParamsIn[i] + "\n";
+			assignRetParams += retParams[i] + " := " + retParamsIn[i] + "\n";
```
Comment (hrkrshnn) [inline libsolidity/codegen/ir/IRGenerator.cpp]: Was this just a typo?

Comment (hrkrshnn) [reply libsolidity/codegen/ir/IRGenerator.cpp]: What about line 441 in the same file? I guess the bug was because it got copied from line 441?

Comment (chriseth) [reply libsolidity/codegen/ir/IRGenerator.cpp]: But there it is correct, right?

Comment (bshastry) [reply libsolidity/codegen/ir/IRGenerator.cpp]: Line 441 lgtm.

**Review by bshastry — DISMISSED** [review]
LGTM

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 1 above)
Comment (hrkrshnn) [inline libsolidity/codegen/ir/IRGenerator.cpp]:
```suggestion
			assignRetParams += retParams[i] + " := " + retParamsIn.at(i) + "\n";
```

or like in line 441, but complementary.

```suggestion
			assignRetParams += retParams[i] + " := " + retParamsIn.back() + "\n";
```

**Review by hrkrshnn — DISMISSED** [review]
Looks good. One optional suggestion.

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/codegen/ir/IRGenerator.cpp]: What about using `retParams.at(i) + " := " + retParamsIn.at(i) + "\n";` in both places?

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**